### DISCUSSION TITLE
How to create value with relation in single query/transaction

### DISCUSSION THREAD

**Discussion by JaLe29** [prisma/prisma-client-go#1222] [Q&A]
I have example schema:
```
model Post {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  published Boolean
  title     String
  content   String?
 
  comments Comment[]
}
 
model Comment {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  content   String
 
  post   Post   @relation(fields: [postID], references: [id])
  postID String
}
```

And I would like to create Post with one comment in single query/transaction. How can I do it?

In documentation, there is a section about create relation - https://goprisma.org/docs/walkthrough/create#create-a-record-with-a-relation but its not helpful.

Same for relation - https://goprisma.org/docs/walkthrough/transactions#setting-relations-with-foreign-keys I would like to use id from first transaction, how?

**Comment by steebchen** [comment]
There's a limitation in the current implementation of the transaction API which means you can't get the ID of a previous result, you have to set the ID (and foreign ID) in advance. In order to support this I would need to implement [interactive transactions](https://github.com/steebchen/prisma-client-go/issues/569)
